QUARKUS-6244.MD

```diff
@@ -0,0 +1,83 @@
+
+JIRA: https://issues.redhat.com/browse/QUARKUS-6244
+
+Upstream issues: https://github.com/quarkusio/quarkus/issues/44278, https://github.com/quarkusio/quarkus/issues/35603
+
+Upstream PR: https://github.com/quarkusio/quarkus/pull/47012, https://github.com/quarkusio/quarkus-quickstarts/pull/1516
+
+Starting with Quarkus 3.22, Quarkus documents support for Jakarta Data 1.0.1 when used with Hibernate ORM extension.
+This feature relies heavily on the `hibernate-jpamodelgen` annotation processor and is implemented almost entirely outside the Quarkus.
+Quarkus itself plans to integrate Jakarta Data support through Panache extension and complement the active record pattern.
+The QUARKUS-6244 itself is about vanilla Hibernate ORM support for Jakarta Data 1.0.1.
+Jakarta Data goals are simplifying data access and their manipulation (e.g. by making sessions stateless), unify API (Jakarta NoSQL and Persistence specs), abstracting layer common for different databases and storage sources and so on.
+That means that we need to test all supported databases.
+We need to verify integration points between Quarkus and Hibernate ORM as the vast majority of upstream test coverage covers vanilla Hibernate ORM.
+There is no clear connection of this feature to OpenShift or any other environment, therefore we just run a smoke test.
+
+## Scope of the testing
+
+Testing must focus on Jakarta Data integration with Quarkus for supported configurations and environments:
+
+- all functionalities shall be tested for PostgreSQL, MariaDB, Oracle, SQL Server and DB2
+- changes to a static metamodel will be tested in a DEV mode
+- smoke test for OpenShift will be run with PostgreSQL (as that is one of the most popular databases)
+- the functionalities won't be tested comprehensively, we need smoke tests for each of them, namely:
+  - Jakarta Data Annotations (e.g. `@Repository`) including their attributes (e.g. the `@Repository` attribute `provider`)
+  - built-In interfaces (e.g. `CrudRepository`)
+  - data retrieval and modifications (insert, update, save, delete for the modification and find, count and exists for the retrieval)
+  - reusing programming model for entity classes from Jakarta Persistence
+  - compatibility with Jakarta Persistence entities and queries using Jakarta Data Query Language
+  - flexible sorting, paging and precedence of sort criteria
+  - type-safety using generated classes from the static metamodel
+  - direct field access for basic fields, embedded fields and association fields
+  - trivial entity associations (not required by specs, but most likely to be tried by users)
+  - entity type inference (either exactly the same type or `Limit`, `Order`, `PageRequest`, `Sort`)
+  - direct access to the connection from a resource accessor method
+  - lifecycle annotations
+- we also need to ensure good integration with functionalities provided by other extensions:
+  - JTA transactions
```

Comment:
I wouldn't mind to test this on OpenShift too, as transactions used to be finnicky on OCP